SpatialGDK/Documentation/content/how-to-use-schema.md

```diff
@@ -6,45 +6,57 @@
+    Run a full scan the first time you generate schema for your project, and whenever you need to generate schema for classes that are not currently loaded by the Editor.<br/>For example: You need to select **Schema (Full Scan)** if you didn't generate schema after adding a new Blueprint to your game, and that Blueprint is no longer open in the Editor.<br/><br/>
 
-* **Generate Schema (Iterative)**
+* **Iterative**
 
-    To generate schema for classes* that are currently loaded by the Editor: <br/>
+    To generate schema for classes that are currently loaded by the Editor: <br/>
 
-    Select **Schema** in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes* that are currently loaded by the Editor, generates the schema files and updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).<br/>
+    Select **Schema** in the [GDK toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes that are currently loaded by the Editor, generates the schema files and updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).<br/>
     ![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button.png)<br/>
-    _Image: In the GDK toolbar in the Unreal Editor, select **Schema**_<br/><br/>
+    _Image: In the GDK toolbar in the Unreal Editor, select **Schema**_<br/>
 
-As the GDK automatically generates all the schema you need, you do not have to write or edit schema manually when using the GDK.
+### Generating schema via the command line
+You can generate schema via the command line using an Unreal commandlet. This method uses a Full Scan (see [Which method to use](#which-method-to-use)). It's useful as part of an automated build system.
 
-> \*Whenever you generate schema, the GDK automatically creates schema for classes with [replicated properties (Unreal documentation)](https://docs.unrealengine.com/en-US/Gameplay/Networking/Actors/Properties) or [RPCs (Unreal documentation)](https://docs.unrealengine.com/en-us/Gameplay/Networking/Actors/RPCs). If a class does not have replicated properties or RPCs, then the GDK does not generate schema for it. 
+To generate schema via the command line, run the following command:
 
-## When to generate schema
+```
+<Path to UE4Editor.exe> <Path to your game's .uproject file> -run=GenerateSchemaAndSnapshots -MapPaths=<MapName1;MapName2;MapName3>
+```
 
-Select **Schema (Full Scan)** if you have: 
+This command generates schema using a Full Scan for all assets, and generates individual snapshots (`<MapName>.snapshot`) for each map. Any specified map paths that end in `/` are interpreted as a directory, and snapshots are generated for each .umap found under these paths.
```

Comment:
1. full scan? (capitalisation consistency)

2. What do we mean by "assets" to we mean all classes which have replicated properties or RPCs? Or do we mean something else entirely here?
"assets" in this context, often means "visual or audio which make up your game like textures and images".
Probs best to keep consistent with the term we are using - unless new term, obvs.

3. What's a snapshot and how can I find out about it? DO we mean SpatialOS snapshots? Or is this something different and if so, why are they called snapshots - help my confused mind!!!? (Super confusing if different!)

Reply:
@padhansell could you take a look at 2 and 3? I imagine that for 3, we're talking about SpatialOS snapshots, in which case I should link to the relevant doc.

Reply:
Yeah for 2) we mean any C++ or blueprint class that has replicated properties or RPCs (so not things like sounds or images) - it might be more correct to say something like "all actor classes". And 3) is indeed referring to SpOS snapshots 😄

Reply:
Cool, thanks. Have made the updates.